airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -201,6 +201,12 @@ def ti_update_state(
 
     if isinstance(ti_patch_payload, TITerminalStatePayload):
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
+        query = query.values(
+            state=ti_patch_payload.state,
+        )
+        if ti_patch_payload.state == State.FAILED:
```

Comment:
tbh we don't need this `if` check I think, we can set `next_method=None, next_kwargs=None` for all `TITerminalStatePayload`

Reply:
I was overriding that in a later PR: https://github.com/apache/airflow/pull/44981